private/bufpkg/bufconfig/write_test.go

```diff
@@ -0,0 +1,96 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufconfig
+
+import (
+	"context"
+	"io"
+	"testing"
+
+	"github.com/bufbuild/buf/private/bufpkg/bufcheck/bufbreaking/bufbreakingconfig"
+	"github.com/bufbuild/buf/private/bufpkg/bufcheck/buflint/buflintconfig"
+	"github.com/bufbuild/buf/private/pkg/storage/storageos"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWriteConfigSuccess(t *testing.T) {
+	storageosProvider := storageos.NewProvider(storageos.ProviderWithSymlinks())
+	readWriteBucket, err := storageosProvider.NewReadWriteBucket(t.TempDir())
+	require.NoError(t, err)
+	writeConfigOptions := []WriteConfigOption{
+		WriteConfigWithVersion(V1Version),
+		WriteConfigWithBreakingConfig(&bufbreakingconfig.Config{
+			Version: V1Version,
+			Use:     []string{"FILE"},
+		}),
+		WriteConfigWithLintConfig(&buflintconfig.Config{
+			Version: V1Version,
+			Use:     []string{"DEFAULT"},
+		}),
+	}
+	err = WriteConfig(context.Background(), readWriteBucket, writeConfigOptions...)
+	require.NoError(t, err)
+	configReadObjectCloser, err := readWriteBucket.Get(context.Background(), ExternalConfigV1FilePath)
+	require.NoError(t, err)
+	configBytes, err := io.ReadAll(configReadObjectCloser)
+	require.NoError(t, err)
+	require.Equal(
+		t,
+		string(configBytes),
+		`version: v1
+breaking:
+  use:
+    - FILE
+lint:
+  use:
+    - DEFAULT
+`,
+	)
+	require.NoError(t, configReadObjectCloser.Close())
+}
+
+func TestWriteConfigMismatchedConfigVersions(t *testing.T) {
+	t.Parallel()
+	t.Run("invalid breaking config", func(t *testing.T) {
+		storageosProvider := storageos.NewProvider(storageos.ProviderWithSymlinks())
+		readWriteBucket, err := storageosProvider.NewReadWriteBucket(t.TempDir())
+		require.NoError(t, err)
+		writeConfigOptions := []WriteConfigOption{
+			WriteConfigWithVersion(V1Beta1Version),
+			WriteConfigWithBreakingConfig(&bufbreakingconfig.Config{
+				Version: V1Version,
+				Use:     []string{"FILE"},
+			}),
+		}
+		err = WriteConfig(context.Background(), readWriteBucket, writeConfigOptions...)
+		require.Error(t, err)
+		require.Equal(t, err.Error(), `invalid version found for breaking config: "v1"`)
+	})
+	t.Run("invalid lint config", func(t *testing.T) {
+		storageosProvider := storageos.NewProvider(storageos.ProviderWithSymlinks())
+		readWriteBucket, err := storageosProvider.NewReadWriteBucket(t.TempDir())
+		require.NoError(t, err)
+		writeConfigOptions := []WriteConfigOption{
+			WriteConfigWithVersion(V1Beta1Version),
+			WriteConfigWithLintConfig(&buflintconfig.Config{
+				Version: V1Version,
+				Use:     []string{"DEFAULT"},
+			}),
+		}
+		err = WriteConfig(context.Background(), readWriteBucket, writeConfigOptions...)
+		require.Error(t, err)
+		require.Equal(t, err.Error(), `invalid version found for lint config: "v1"`)
```

Comment:
nit: we should say what we expect the version to be - otherwise the error message is confusing (`v1` is a valid version).

Reply:
Ah yeah, that's a good call.